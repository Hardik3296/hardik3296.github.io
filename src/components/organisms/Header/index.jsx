import React from "react";
import { LinkedIn, GitHub, Mail } from "@material-ui/icons"
import image from "../../../assets/images/example.jpg"
import styles from "./styles.module.scss";


const Header = () => {
   return (
      <div className={styles.container}>
         <img src={image} className={styles.image} alt="" />
         <p>Hardik Gaur</p>
         <div className={styles.iconDiv}>
            <a className={styles.icon} href="https://www.linkedin.com/in/hardik-gaur-135891122/" target="blank"><LinkedIn style={{ color: "#0077b5", fontSize: 35 }} /></a>
            <a href="mailto:hardik3296@outlook.com" className={styles.icon}><Mail style={{ color: "#6c63ff", fontSize: 35 }} /></a>
            <a href="https://github.com/Hardik3296" target="blank" className={styles.icon}><GitHub style={{ color: "#000000", fontSize: 35 }} /></a>
         </div>
      </div>
   );
}

export default Header;