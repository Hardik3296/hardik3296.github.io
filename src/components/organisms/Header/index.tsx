import { Dispatch, SetStateAction, useContext } from "react";
import SocialMedia from "../../molecules/Social-Media";
import image from "../../../assets/images/example.jpg"
import styles from "./styles.module.scss";
import { ThemeContext, ThemeInterface } from "../../../utils/contexts/ThemeContext";


const Header = (): JSX.Element => {

   const [theme, _] = useContext<[ThemeInterface, Dispatch<SetStateAction<ThemeInterface>>]>(ThemeContext);
   return (
      <div className={styles.container} style={theme.header}>
         <img src={image} className={styles.image} alt="" />
         <p style={theme.text}>Hardik Gaur</p>
         <div className={styles.iconDiv}>
            <SocialMedia size={35} icons={theme.icons} />
         </div>
      </div>
   );
}

export default Header;