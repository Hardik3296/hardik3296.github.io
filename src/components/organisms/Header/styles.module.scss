@import "../../../fonts.scss";

.container{
   position: sticky;
   height:10vh;
   z-index:2;
   padding:8px 8px;
   top:0;
   display:flex;
   flex-direction: row;
   justify-content: space-evenly;
   align-items: center;
   &>p{
      font-family: ArchivoMedium;
      font-size: xx-large;
   }
   .image{
      height:10vh;
      width:10vh;
      border-radius: 10vh;
   }
}