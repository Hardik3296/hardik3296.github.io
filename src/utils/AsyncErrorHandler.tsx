import React from "react";

const useAsyncError = () => {
   const [_error, setError] = React.useState();
   return React.useCallback(
      e => {
         setError(() => {
            throw e;
         });
      },
      [setError],
   );
};

export default useAsyncError;