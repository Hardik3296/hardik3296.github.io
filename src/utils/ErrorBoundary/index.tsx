import { Component, ReactNode } from "react";
import styles from "./styles.module.scss";

interface MyProps {
   textcolor: string,
   children: ReactNode,
}

class ErrorBoundary extends Component<MyProps, {}> {

   state = {
      hasError: false
   }

   static getDerivedStateFromError() {
      return { hasError: true }
   }

   componentDidCatch(error: any, errorInfo: any) {
   }
   render() {
      if (this.state.hasError)
         return <div className={styles.errorDiv}>
            <p style={{ color: this.props.textcolor }}>Something went wrong while fetching the repositories!!</p>
         </div>
      else
         return this.props.children;
   }
}

export default ErrorBoundary;