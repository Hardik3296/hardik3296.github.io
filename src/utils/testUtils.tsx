import { ShallowWrapper } from "enzyme"

export const findAttributeByTestId = (component:ShallowWrapper, attr:string):ShallowWrapper=>{
   return component.find(`[test-id='${attr}']`);
}