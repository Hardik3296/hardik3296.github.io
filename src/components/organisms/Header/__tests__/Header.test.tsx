import { shallow, ShallowWrapper } from "enzyme";
import { findAttributeByTestId } from "../../../../utils/testUtils";
import renderer from "react-test-renderer";
import Header from "..";

const renderComponent = (): ShallowWrapper => {
   return shallow(<Header />)
}

describe("Header Component", () => {
   let wrapper: ShallowWrapper;

   beforeEach(() => {
      wrapper = renderComponent();
   });

   it("Should have my pic", () => {
      const component = findAttributeByTestId(wrapper, "image");
      expect(component.length).toBe(1);
   });

   it("Should have my name", () => {
      const component = findAttributeByTestId(wrapper, "name");
      expect(component.length).toBe(1);
   });

   it("Should have social-media", () => {
      const component = findAttributeByTestId(wrapper, "social-media");
      expect(component.length).toBe(1);
   });

   it("Should have menu", () => {
      const component = findAttributeByTestId(wrapper, "menu");
      expect(component.length).toBe(1);
   });

   it("Should render correctly", () => {
      const component = renderer.create(<Header />).toJSON();
      expect(component).toMatchSnapshot();
   })

})