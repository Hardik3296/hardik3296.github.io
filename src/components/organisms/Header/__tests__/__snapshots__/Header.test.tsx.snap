// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Should render correctly 1`] = `
<div
  className="container"
  style={
    Object {
      "backgroundColor": "#f2f2f2",
      "borderBottom": "1px solid rgba( 255, 255, 255, 0.18)",
      "boxShadow": "8px 0px 10px 4px rgba(31, 38, 135, 0.37)",
    }
  }
>
  <img
    alt="My pic"
    className="image"
    src="me.webp"
    test-id="image"
  />
  <p
    className="name"
    style={
      Object {
        "color": "#000000",
      }
    }
    test-id="name"
  >
    Hardik Gaur
  </p>
  <div
    className="iconDiv"
    test-id="social-media"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root icon"
      focusable="false"
      onClick={[Function]}
      style={
        Object {
          "color": "#0077b5",
          "fontSize": 35,
          "marginRight": 20,
        }
      }
      test-id="linkedin"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
      />
    </svg>
    <svg
      className="icon"
      color="#6c63ff"
      height={35}
      onClick={[Function]}
      test-id="medium"
      viewBox="0 0 100 100"
      width={35}
    >
      <path
        d="M10 8a2 2 0 0 0-2 2v80a2 2 0 0 0 2 2h80a2 2 0 0 0 2-2V10a2 2 0 0 0-2-2zm2 4h76v76H12zm10 14v1l3.715 5c.183.219.285.492.285.777V63c0 .441-.152.871-.434 1.215L20 71.094V72h16v-.906l-5.57-6.918a1.935 1.935 0 0 1-.43-1.211V36.75L46.61 72h3.085l.012-.04L64 35.759V65.5a1.2 1.2 0 0 1-.36.855L58 71.094V72h22v-.906l-3.64-4.739A1.2 1.2 0 0 1 76 65.5V32.25c0-.32.129-.629.36-.855L80 27v-1H63.55L51.993 55.285 38.195 26zm9.242 4h4.414l14.32 30.39-2.011 5.098h-.004zM68 30h4.516A5.189 5.189 0 0 0 72 32.25V65.5c0 .875.227 1.738.645 2.5h-5.29A5.212 5.212 0 0 0 68 65.5zm0 0"
        style={
          Object {
            "fill": "#6c63ff",
            "fillOpacity": 1,
            "fillRule": "nonzero",
            "stroke": "none",
          }
        }
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root icon"
      focusable="false"
      onClick={[Function]}
      style={
        Object {
          "color": "#6c63ff",
          "fontSize": 35,
          "marginLeft": 20,
          "marginRight": 20,
        }
      }
      test-id="mail"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
      />
    </svg>
    <svg
      className="icon"
      color="#6c63ff"
      height={35}
      onClick={[Function]}
      test-id="geeksforgeeks"
      viewBox="0 0 96 96"
      width={35}
    >
      <path
        d="M58.07 48a30.51 30.51 0 0 1-.07-2c0-12.16 5.719-20 14-20 6.82 0 12.66 5.324 14 14h4l.004-18H86s-1.066 3.012-2 2.32c-3.797-2.133-7.445-2.265-12.047-2.265C60.352 22.055 50 32.52 50 45.84c0 .726.043 1.445.098 2.16h-4.2c.059-.715.102-1.434.102-2.16 0-13.32-10.352-23.785-21.953-23.785-4.602 0-8.25.133-12.047 2.265-.934.692-2-2.32-2-2.32H5.996L6 40h4c1.34-8.676 7.18-14 14-14 8.281 0 14 7.84 14 20 0 .68-.027 1.344-.07 2H0v4h2.02c2.164 0 3.953 1.723 3.996 3.887C6.094 59.578 6.449 64.012 8 66c2.54 3.25 6 6 16 6 10.043 0 19.84-9.055 22-20h4c2.16 10.945 11.957 20 22 20 10 0 13.46-2.75 16-6 1.55-1.988 1.906-6.422 1.984-10.113C90.027 53.723 91.816 52 93.98 52H96v-4zM22.953 67.46c-3.21 0-8.308-.01-8.953-3.46-.219-1.164-.184-5.055-.113-8.094a3.999 3.999 0 0 1 4-3.906H38c-1.59 9.563-7.969 15.46-15.047 15.46zM82 64c-.645 3.45-5.742 3.46-8.953 3.46C65.973 67.46 59.59 61.564 58 52h20.113a3.999 3.999 0 0 1 4 3.906c.07 3.04.106 6.93-.113 8.094zm0 0"
        style={
          Object {
            "fill": "#6c63ff",
            "fillOpacity": 1,
            "fillRule": "nonzero",
            "stroke": "none",
          }
        }
      />
    </svg>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root icon"
      focusable="false"
      onClick={[Function]}
      style={
        Object {
          "color": "#333333",
          "fontSize": 35,
          "marginLeft": 20,
        }
      }
      test-id="github"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
      />
    </svg>
  </div>
  <div
    className="drawerDiv"
    test-id="menu"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      onClick={[Function]}
      style={
        Object {
          "color": "#000000",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
</div>
`;
